adminpage/views.py

```diff
@@ -1,3 +1,272 @@
 # Create your views here.
+
+
+class Login(APIView):
+
+    def get(self):
+        if not self.request.user.is_authenticated():
+            raise ValidateError('User not logged in')
+        return 0
+
+    def post(self):
+        self.check_input('username', 'password')
+        username = self.input['username']
+        password = self.input['password']
+        user = authenticate(username=username, password=password)
+
+        if not username:
+            raise ValidateError('Username is empty！')
+        if not password:
+            raise ValidateError('Password is empty！')
+        if not User.objects.filter(username=username):
+            raise ValidateError('Username does not exist！')
+        if not user:
+            raise ValidateError('Wrong password！')
+
+        if user.is_active:
+            login(self.request, user)
+            return 0
+
+
+class Logout(APIView):
+
+    def post(self):
+        if self.request.user.is_authenticated():
+            logout(self.request)
+        else:
+            raise ValidateError('Logout failed！User has not logged in.')
+
+
+class ActivityList(APIView):
+
+    def get(self):
+        if not self.request.user.is_authenticated():
+            raise ValidateError('User not logged in！')
+
+        activityList = Activity.objects.all()
+        data = []
+        for item in activityList:
+            if item.status >= 0:
+                temp = {
+                    'id': item.id,
+                    'name': item.name,
+                    'description': item.description,
+                    'startTime': int(time.mktime(item.start_time.timetuple())),
+                    'endTime': int(time.mktime(item.end_time.timetuple())),
+                    'place': item.place,
+                    'bookStart': int(time.mktime(item.book_start.timetuple())),
+                    'bookEnd': int(time.mktime(item.book_end.timetuple())),
+                    'currentTime': int(time.time()),
+                    'status': item.status,
+                }
+                data.append(temp)
+            else:
+                continue
+        return data
+
+
+class ActivityDelete(APIView):
+
+    def post(self):
+        self.check_input('id')
+        activity = Activity.get_by_id(self.input['id'])
+        activity.delete()
+
+
+class ActivityCreate(APIView):
+    def post(self):
+        self.check_input('name', 'key', 'place', 'description', 'picUrl', 'startTime', 'endTime', 'bookStart', 'bookEnd', 'totalTickets', 'status')
+        if not self.request.user.is_authenticated():
+            raise ValidateError('User not logged in！')
+        new_activity = Activity(
+            name = self.input['name'],
+            key = self.input['key'],
+            place = self.input['place'],
+            description = self.input['description'],
+            pic_url = self.input['picUrl'],
+            start_time = datetime.strptime(self.input['startTime'], "%Y-%m-%dT%H:%M:%S.%fZ") + timedelta(hours=8),
```

Comment:
if u have some errors in time, maybe change the config of MySQL(or other db) is better